bk-sshd/ROOTFS/opt/sage/beekeeper/register/register.sh

```diff
@@ -1,34 +1,47 @@
-#!/bin/bash -e
+#!/bin/bash
 # ANL:waggle-license
 #  This file is part of the Waggle Platform.  Please see the file
 #  LICENSE.waggle.txt for the legal details of the copyright and software
 #  license.  For more details on the Waggle project, visit:
 #           http://www.wa8.gl
 # ANL:waggle-license
 
+# NOTE(sean) This script forwards JSON output directly to a consumer, so do not write anything else to stdout!
+
+set -eu
 
 bk_register_url=http://bk-api:5000
 if [ -e /config/BEEKEEPER_REGISTER_API ]; then
   # env variables cannot be passed to this script, thus we read config from file
   bk_register_url=$(cat /config/BEEKEEPER_REGISTER_API)
 fi
 
-#echo "bk_register_url: ${bk_register_url}"
-
-run_command() {
-  input=($1)
-
-  case ${input[0]} in
-    register)
+# process args baked into authorized command
+beehive=""
 
-      curl -s -X GET "${bk_register_url}/register?id=${input[1]}"
-
-      ;;
-    *)
-      echo "invalid command"
-      exit 1
-      ;;
+while getopts "b:" opt; do
+  case "${opt}" in
+    b) beehive="${OPTARG}" ;;
   esac
-}
-
-run_command "$SSH_ORIGINAL_COMMAND"
+done
+
+# process args provided via ssh
+user_args=(${SSH_ORIGINAL_COMMAND})
+command="${user_args[0]}"
+
+case "${command}" in
+  register)
+    node="${user_args[1]}"
+    params="node_id=${node}"
+
+    if [ ! -z "${beehive}" ]; then
```

Comment:
The API manages the "default beehive" var. Omitting the beehive_id causes the API to use that, so we don't need to also track that here.

Reply:
Could we have just sent an empty `${beehive}`?

basically does this work ??

```
/register?node_id=NODE123&beehive_id=
```

Reply:
I personally like omitting things if we want the default, as that (usually) allows existing call to continue working, but we could have both behaviors. (Omitting or empty value fallback to the default.)

Reply:
what you have is fine. just was curious.